saleor/graphql/order/bulk_mutations/order_bulk_create.py

```diff
@@ -1233,178 +1576,248 @@ def create_single_fulfillment(
             if order_line := cls.create_single_order_line(
                 order_line_input,
-                order_instance,
+                order_data,
                 object_storage,
                 order_input,
-                order.lines_errors,
+                order_line_index,
             ):
-                order.lines.append(order_line)
+                order_data.lines.append(order_line)
             else:
-                order.is_critical_error = True
+                order_data.is_critical_error = True
+            order_line_index += 1
 
-        if order.is_critical_error:
-            return order
+        if order_data.is_critical_error:
+            order_data.order = None
+            return order_data
 
         # calculate order amounts
         order_amounts = cls.make_order_calculations(
             delivery_method,
-            order.all_order_lines,
-            instances.channel,
+            order_data.all_order_lines,
+            order_data.channel,
             delivery_input,
             object_storage,
         )
 
         # create notes
         if notes_input := order_input.get("notes"):
+            note_index = 0
             for note_input in notes_input:
                 if note := cls.create_single_note(
-                    note_input, order_instance, object_storage, order.notes_errors
+                    note_input, order_data, object_storage, note_index
                 ):
-                    order.notes.append(note)
+                    order_data.notes.append(note)
+                note_index += 1
 
         # create fulfillments
         if fulfillments_input := order_input.get("fulfillments"):
+            fulfillment_index = 0
             for fulfillment_input in fulfillments_input:
                 if fulfillment := cls.create_single_fulfillment(
                     fulfillment_input,
-                    order.lines,
-                    order_instance,
+                    order_data.lines,
+                    order_data,
                     object_storage,
-                    order.fulfillments_errors,
+                    fulfillment_index,
                 ):
-                    order.fulfillments.append(fulfillment)
+                    order_data.fulfillments.append(fulfillment)
                 else:
-                    order.is_critical_error = True
-            if order.is_critical_error:
-                return order
+                    order_data.is_critical_error = True
+                fulfillment_index += 1
+            if order_data.is_critical_error:
+                order_data.order = None
+                return order_data
 
         # create transactions
         if transactions_input := order_input.get("transactions"):
+            transaction_index = 0
             for transaction_input in transactions_input:
                 try:
                     transaction = TransactionCreate.prepare_transaction_item_for_order(
```

Comment:
I am not convinced of having this method inside `TransactionCreate`. As far as I see, it is used only in this file.

Reply:
Also in case of adding any transaction that is attached to the order. You need to proceed of few steps that are mandatory. You can check `TransactionCreate.perform_mutation`.
```
new_transaction = cls.create_transaction(transaction_data, user=user, app=app)
if money_data:
    create_manual_adjustment_events(
        transaction=new_transaction, money_data=money_data, user=user, app=app
    )
    recalculate_transaction_amounts(new_transaction)
if transaction_data.get("order_id") and money_data:
    order = cast(order_models.Order, new_transaction.order)
    cls.update_order(order, money_data, update_search_vector=True)

    order_info = fetch_order_info(order)
    order_transaction_updated(
        order_info=order_info,
        transaction_item=new_transaction,
        manager=manager,
        user=user,
        app=app,
        previous_authorized_value=Decimal(0),
        previous_charged_value=Decimal(0),
    )
```

Reply:
`create_manual_adjustment_events` creates the events that will be used to re-calcualte the transaction's amounts. The amounts provided in `transaction` are based on the events' amounts. So without adding events, any update on imported transaction  will set amounts to 0.

Reply:
`order_transaction_updated` probably should be skipped as it triggers the webhooks like order_fully_piad